src/skydb.js

```diff
@@ -0,0 +1,245 @@
+ */
+const nodejs_db_setJSON = async function (privateKey, dataKey, json, customOptions = {}) {
+  const opts = { ...defaultSkydbOptions, ...this.customOptions, ...customOptions };
+
+  const publicKey = getPublicKeyfromPrivateKey(privateKey);
+  const { entry, dataLink } = await getOrCreateRegistryEntry(this, publicKey, dataKey, json, opts);
+
+  // Update the registry.
+  await this.registry.setEntry(privateKey, entry, defaultSetEntryOptions);
+
+  return { data: json, dataLink: dataLink };
+};
+
+/**
+ * Skydb v1
+ * Gets the registry entry and data link or creates the entry if it doesn't exist.
+ *
+ * @param client - The Skynet client.
+ * @param publicKey - The user public key.
+ * @param dataKey - The dat akey.
+ * @param json - The JSON to set.
+ * @param [customOptions] - Additional settings that can optionally be set.
+ * @returns - The registry entry and corresponding data link.
+ * @throws - Will throw if the revision is already the maximum value.
+ */
+const getOrCreateRegistryEntry = async function (client, publicKey, dataKey, json, customOptions = {}) {
+  const opts = { ...defaultSkydbOptions, ...client.customOptions, ...customOptions };
+
+  // Set the hidden _data and _v fields.
+  const skynetJson = await buildSkynetJsonObject(json);
+  const fullData = JSON.stringify(skynetJson);
+
+  // uploads in-memory data to skynet
+  const { skylink, shortskylink } = await uploadJSONdata(client, fullData, dataKey, opts);
+
+  // Fetch the current value to find out the revision.
+  const signedEntry = await client.registry.getEntry(publicKey, dataKey, defaultGetEntryOptions);
+
+  const revision = getNextRevisionFromEntry(signedEntry.entry);
+
+  // Build the registry entry.
+  const rawDataLink = decodeSkylinkBase64(shortskylink);
+  if (rawDataLink.length !== raw_skylink_size) {
+    throw new Error("RawDataLink is not 34 bytes long.");
+  }
+
+  const entry = {
+    dataKey,
+    data: rawDataLink,
+    revision,
+  };
+
+  return { entry: entry, dataLink: skylink };
+};
+
+/**
+ * Skydb v1
+ * Gets the next revision from a returned entry (or 0 if the entry was not found).
+ *
+ * @param entry - The returned registry entry.
+ * @returns - The revision.
+ * @throws - Will throw if the next revision would be beyond the maximum allowed value.
+ */
+const getNextRevisionFromEntry = function (entry) {
+  let revision;
+  if (entry === null) {
+    revision = BigInt(0);
+  } else {
+    revision = entry.revision + BigInt(1);
+  }
+  // Throw if the revision is already the maximum value.
+  if (revision > max_revision) {
+    throw new Error("Current entry already has maximum allowed revision, could not update the entry");
+  }
+  return revision;
+};
+
+/**
+ *
+ * Seection for Skydb v2 for node.js.
+ *
+ */
+
+/**
+ * Skydb v2
+ * Sets a JSON object at the registry entry corresponding to the publicKey and
+ * dataKey.
```

Comment:
```suggestion
 * Sets a JSON object at the registry entry corresponding to the privateKey and
 * dataKey using SkyDB V2.
```

Reply:
I integrated it extrat so that developers and I have a better overview of which function belongs to which skydb version.

At the end, just before it's released, when we're done, I can take everything out and see how big it gets, then maybe split it into two files.